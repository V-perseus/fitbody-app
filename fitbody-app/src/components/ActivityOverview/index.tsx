import ActivityOverview from './ActivityOverview'

export default ActivityOverview
