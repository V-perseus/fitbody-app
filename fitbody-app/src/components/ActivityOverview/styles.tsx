import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  contentContainer: {
    paddingHorizontal: 31,
    paddingTop: 16,
  },
})

export default styles
