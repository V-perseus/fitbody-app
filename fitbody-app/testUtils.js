import React from 'react'
import { render as rtlRender } from '@testing-library/react-native'
import { configureStore } from '@reduxjs/toolkit'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { Provider } from 'react-redux'

// Import your own reducer
import { appReducer } from './src/store'
import { ModalStackScreens } from './src/config/routes/loggedInStack'

// custom render method override
function render(ui, { preloadedState, store = configureStore({ reducer: appReducer, preloadedState }), ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }

// custom navigation mock
const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
  preloadedState,
  store = configureStore({ reducer: appReducer, preloadedState }),
}) => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="MockedScreen" component={component} initialParams={params} />
          <Stack.Screen name="Modals" component={ModalStackScreens} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  )
}
