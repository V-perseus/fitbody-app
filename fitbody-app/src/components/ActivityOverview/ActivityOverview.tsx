import React from 'react'
import { FlatList } from 'react-native'
import { useSelector } from 'react-redux'

// styles
import styles from './styles'

// Services
import { activitiesSelector } from '../../data/journal/selectors'

// Components
import { ActivityItem } from '../ActivityItem/ActivityItem'
import { IActivity } from '../../data/journal/types'

interface IActivityOverviewProps {
  myActivities: number[]
}
export default function ActivityOverview({ myActivities }: IActivityOverviewProps) {
  const activities = useSelector(activitiesSelector)

  function keyExtractor(item: number) {
    return `activity-${item}`
  }

  function renderItem({ item, index }: { item: number; index: number }) {
    if (item < 0) {
      return null
    }

    const activity = activities.find((act) => act.id === item) || ({ name: 'Unknown', icon_key: 'exercised' } as IActivity)

    return <ActivityItem name={activity.name} iconKey={activity.icon_key} index={index} />
  }

  return (
    <FlatList
      contentContainerStyle={styles.contentContainer}
      data={myActivities}
      renderItem={renderItem}
      keyExtractor={keyExtractor}
      horizontal
    />
  )
}
