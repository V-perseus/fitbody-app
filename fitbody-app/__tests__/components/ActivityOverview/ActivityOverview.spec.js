import React from 'react'
import { render, cleanup } from '../../../testUtils'

import ActivityOverview from '../../../src/components/ActivityOverview/ActivityOverview'

afterEach(cleanup)

const myActivities = [
  2, // exercised
  3, // ate healthy
  4, // meal prepped
  5, // drank 3-4 liters
  6, // took a rest
  7, // sletp 6-8
]

describe('ActivityOverview', () => {
  it('renders snapshot correctly', () => {
    const tree = render(<ActivityOverview activities={myActivities} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders the exercised activities', () => {
    const { getByTestId } = render(<ActivityOverview myActivities={myActivities} />)
    expect(getByTestId('activity_item_0')).not.toBeNull()
    expect(getByTestId('activity_item_5')).not.toBeNull()
  })
})
