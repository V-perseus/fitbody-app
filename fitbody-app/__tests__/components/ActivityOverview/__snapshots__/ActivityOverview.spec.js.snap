// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityOverview renders snapshot correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingHorizontal": 31,
      "paddingTop": 16,
    }
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View />
</RCTScrollView>
`;
